### PR TITLE
Update kaniko image v1.0.0 -> v1.5.1

### PR DESCRIPTION
**Pull request by ImJasonH**
Since 1.0.0 a lot of multi-arch improvements have been made, as well as
performance improvements and bug fixes

See full changelog: https://github.com/GoogleContainerTools/kaniko/blob/master/CHANGELOG.md

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [y] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [y] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [y] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [y] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [y] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [y] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [y] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [y] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [y] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/kaniko/0.2/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+>standard Kubernetes cluster.
+> - [Kaniko website](https://github.com/GoogleCloudPlatform/kaniko)
+
+kaniko is meant to be run as an image, `gcr.io/kaniko-project/executor:v1.5.1`. This
+makes it a perfect tool to be part of Tekton.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/kaniko/0.1/kaniko.yaml
```
Comment (PuneetPunamiya) [inline task/kaniko/0.2/README.md:18]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/kaniko/0.2/kaniko.yaml
```

File: task/kaniko/0.2/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+## Parameters
+
+* **IMAGE**: The name (reference) of the image to build.
+* **DOCKERFILE**: The path to the `Dockerfile` to execute (_default:_ `./Dockerfile`)
+* **CONTEXT**: The build context used by Kaniko (_default:_ `./`)
+
```
[the remark this answers is not in the document]
Comment (ImJasonH) [reply task/kaniko/0.2/README.md]: Done!

File: task/kaniko/0.2/kaniko.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    env:
+    - name: DOCKER_CONFIG
+      value: /tekton/home/.docker
+    command:
+    - /kaniko/executor
+    - $(params.EXTRA_ARGS)
+    - --dockerfile=$(params.DOCKERFILE)
+    - --context=$(workspaces.source.path)/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
+    - --destination=$(params.IMAGE)
+    - --oci-layout-path=$(workspaces.source.path)/$(params.CONTEXT)/image-digest
```
Comment (06kellyjac) [inline task/kaniko/0.2/kaniko.yaml:54]: Entrypoint is already `/kaniko/executor` so can just use args

```suggestion
    args:
    - $(params.EXTRA_ARGS)
    - --dockerfile=$(params.DOCKERFILE)
    - --context=$(workspaces.source.path)/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
    - --destination=$(params.IMAGE)
    - --oci-layout-path=$(workspaces.source.path)/$(params.CONTEXT)/image-digest
```

Comment (ImJasonH) [reply task/kaniko/0.2/kaniko.yaml:54]: Done.

**Review by vinamra28 — APPROVED** [review]
/meow

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Comment by zhangtbj** [comment]
Hi Jason,

I quick question, what is the rule of the version in the Tekton catalog?
For Kaniko as an example, it  provides `0.1` and `0.2` versions in the kaniko folder, but `0.2` uses the Kaniko `1.5.1` version. I think it is a little confusing for end-user.

So:
- Dose Tekton need to keep multiple versions for the tasks? or one latest version is enough?
- If prefer to keep multiple versions, should Tekton catalog use a related version as the folder name?

Thanks! :)

**Comment by vdemeester** [comment]
@zhangtbj See [TEP-0003](https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md) on how the catalog is organized 👼🏼

**Comment by zhangtbj** [comment]
Oh I see, thanks for the TEP document! @vdemeester ! :)

**Pull request merged**